src/ot1d.jl

```diff
@@ -0,0 +1,283 @@
+"""
+otCost1d(c,μ::Distributions.UnivariateDistribution,
+           ν::Distributions.UnivariateDistribution)
+
+Calculates the Optimal Transport Cost between μ to ν, where
+they are 1-Dimensional distributions and the cost
+function is of the form ``c(x,y) = h(|x-y|)`` such that
+``h`` is a convex function.
+
+Parameters:\n
+c: The cost function, which should be of the form ``c(x,y) = h(abs(x-y))``
+where `h` is a convex function.
+μ: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+ν: 1-D distribution (e.g. `μ = Distributions.Normal(0,1)`)\n
+"""
+function otCost1d(c,μ::Distributions.UnivariateDistribution,ν::Distributions.UnivariateDistribution)
+    g(μ,ν,x) = c(quantile(μ,x),quantile(ν,x))
+    f(x) = g(μ,ν,x)
+    return quadgk(f,0,1)[1]
```

Comment:
It is not necessary to create all these functions:
```suggestion
    integral, _ = quadgk(0, 1) do x
        return c(quantile(μ, x), quantile(ν, x))
    end
    return integral
```